modules/client/src/connext.ts

```diff
@@ -1023,6 +1116,37 @@ export class ConnextInternal extends ConnextChannel {
     return appInfo[0];
   };
 
+  public matchTx = (
+    givenTransaction: Transaction,
+    expected: CFCoreTypes.MinimalTransaction,
+  ): boolean => {
+    return (
+      givenTransaction.to === expected.to &&
+      bigNumberify(givenTransaction.value).eq(expected.value) &&
+      givenTransaction.data === expected.data
+    );
+  };
+
+  public retryNodeSubmittedWithdrawal = async (): Promise<void> => {
+    const val = await this.getLatestNodeSubmittedWithdrawal();
+    if (!val) {
+      this.logger.error(`No transaction found to retry`);
+      return;
+    }
+    let { retry, tx } = val;
+    retry += 1;
+    if (retry >= 3) {
+      const msg = `Tried to have node submit withdrawal 3 times and it did not work, try submitting from wallet.`;
+      this.logger.error(msg);
+      // TODO: make this submit from wallet :)
+      // but this is weird, could take a while and may have gas issues.
+      // may not be the best way to do this
```

Comment:
Thinking the best way to do this is not to automatically do it, but inform the user and let them do it if they want to spend the gas.

Reply:
Not sure how best to do this^ we already log the error to console. It's kind of hard to pass back a return value here

Reply:
Well, we are throwing the error, so we can possibly put the channel in some error state that means it requires user interaction.

Reply:
Let's put this in a separate issue for now.